src/client/elements/pageNavNewAPI.ml

```diff
@@ -0,0 +1,157 @@
+open Nes
+open Dancelor_common_model
+open Dancelor_client_html.NewAPI
+
+type state = {
+  current_page : int; (* first page is [1] *)
+  entries_per_page : int;
+  number_of_entries : int;
+}
+
+let number_of_pages state =
+  (state.number_of_entries
+   + state.entries_per_page
+   - 1)
+  / state.entries_per_page
+
+let current_pagination state =
+  (* NOTE: Our page numbers start at 1. *)
+  Pagination.{
+    start = (state.current_page - 1) * state.entries_per_page;
+    end_ = min (state.current_page * state.entries_per_page) state.number_of_entries;
+  }
+
+type t = {
+  signal : state React.signal;
+  setter : state -> unit;
+  updater : (state -> state) -> unit;
+}
+
+let create ~number_of_entries ~entries_per_page =
+  let initial = {
+    current_page = 1;
+    entries_per_page;
+    number_of_entries = 0;
```

Comment:
I am not sure I understand well why `number_of_entries` and `entries_per_page` are handled differently here.

Reply:
The `number_of_entries` argument is an Lwt promise that will eventually return the number of entries. So in the meantime we set it to 0 and when the promise returns we replace that value. `entries_per_page` is known statically. Would renaming arguments help? I'm not a big fan of `*_lwt` arguments when the type already says it but if it avoids the confusion I'm down.

Reply:
Alternatively I could just add the type of the function.

Reply:
Annotating with types could be useful

Reply:
Done in e2fdabc.